### PR TITLE
Add unit tests for `parse_deploy_params`.

### LINKED ISSUE #1670: Write unit tests for parse_deploy_params
We need tests for this function: https://github.com/casper-network/casper-node/blob/dev/client/lib/parsing.rs#L335-L358

### PR DESCRIPTION
**Pull request by nilz3ro**
# About These Changes
* Adds unit tests for `parse_deploy_params`
* closes #1670

### CODE REVIEW

File: client/lib/parsing.rs [hunk 1]
```diff
@@ -965,6 +966,119 @@ mod tests {
         );
     }
 
+    #[test]
+    fn should_parse_valid_deploy_params() {
+        // create secret key file in tempdir.
+        let keys_dir = tempdir().expect("Failed to create temp dir.");
+        let secret_key_path = keys_dir.path().join("key.pem");
+        let secret_key_path_clone = secret_key_path.clone();
+        let secret_key_path_str = secret_key_path_clone.to_str().unwrap();
```
Comment (darthsiroftardis) [inline client/lib/parsing.rs:975]: Maybe merge this into one line:
`let secret_key_path_str = secret_key_path.clone().to_str()?;`

**Review by darthsiroftardis — APPROVED** [review]
One nitpick, but looks good!

File: client/lib/parsing.rs (hunk 1 above)
Comment (darthsiroftardis) [reply client/lib/parsing.rs:975]: Could you try `let secret_key_path_str = secret_key_path.to_str().unwrap()` directly, see if the compiler complains about the value being moved.

**Comment by nilz3ro** [comment]
bors merge

**Pull request merged**